### ISSUE TITLE
Unitest is too Slow

### ISSUE THREAD

**Issue by allwefantasy** [databricks/spark-deep-learning#65]
Run all Unitest take almost 30m. I guess we may optimize the time.

**Comment by thunterdb** [comment]
Indeed, this is a pain point for developers. We have not had a chance to look into it yet, but for starters, running cProfile and looking at the hot calls may expose some obvious issues.

**Comment by thunterdb** [comment]
A quick analysis using cProfile shows that for the test `test_keras_consistency`, out of 50 seconds, about 40 seconds are spent loading the inception model in Keras. While this is a Keras issue, we can at least mitigate the issue by loading the models once when running the tests and keeping them in memory. We are trading speed for memory, so this may cause some other issues if we add a lot of different models in the tests.

cc @sueann
